packages/@react-spectrum/color/src/ColorEditor.tsx

```diff
@@ -47,7 +53,7 @@ function ColorEditor(props: SpectrumColorEditorProps, ref: DOMRef<HTMLDivElement
         {format === 'hex'
           ? <ColorField isQuiet width="size-1000" aria-label={formatter.format('hex')} />
           : getColorChannels(format).map(channel => (
-            <ColorField key={channel} colorSpace={format === 'hex' ? 'rgb' : format} channel={channel} isQuiet width="size-400" flex UNSAFE_style={{'--spectrum-textfield-min-width': 0} as CSSProperties} />
+            <ColorField key={channel} colorSpace={resolvedFormat} channel={channel} isQuiet width="size-400" flex UNSAFE_style={{'--spectrum-textfield-min-width': 0} as CSSProperties} />
```

Comment:
i feel like this is a bug, but I haven't managed to reproduce it in the TS playground

https://www.typescriptlang.org/play/?#code/C4TwDgpgBAwg9gGzgJwMpgIYGNoF4oDkyA5gEYFQA+hAFgM4IEDcAUCwhMFBgFyyIp02aNQI0IADwr4xk5mwCWAMygAKDFFxbacgJRQA3iyjdNhEuVYBfNhy7I+8JGkw5WyMxlbtOUUo4EXYSodKTNZKVY7KGQAJgDnIRwzUk1tCIoAfnMyCj5SViA

Reply:
```suggestion
            <ColorField key={channel} colorSpace={format} channel={channel} isQuiet width="size-400" flex UNSAFE_style={{'--spectrum-textfield-min-width': 0} as CSSProperties} />
```

Reply:
`'hex'` has already been excluded by the ternary condition, hence the type of `format` has been narrowed to `ColorSpace`.

Reply:
well, i feel silly now, thank you for pointing that out